GCEWindowsAgent/wsfc.go

```diff
@@ -0,0 +1,273 @@
+//  Copyright 2017 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package main
+
+import (
+	"net"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/GoogleCloudPlatform/compute-image-windows/logger"
+)
+
+const wsfcDefaultAgentPort = "59998"
+
+type agentState int
+
+// Enum for agentState
+const (
+	running agentState = iota
+	stopped
+)
+
+var (
+	once          sync.Once
+	agentInstance *wsfcAgent
+)
+
+type wsfcManager struct {
+	agentNewState agentState
+	agentNewPort  string
+	agent         healthAgent
+}
+
+// Create new wsfcManager based on metadata
+func newWsfcManager(newMetadata *metadataJSON) *wsfcManager {
+	newState := stopped
+	if newMetadata.Instance.Attributes.EnableWSFC ||
```

Comment:
WSFCAddresses always takes priority as an advanced setting. Same for the address manager will always respect this setting and ignore enable-wsfc if it is set.